fix: Remove `mustCallSuper` from `onComponentTypeCheck` (#2561)
flame-engine/flame: doc/flame/collision_detection.md

      // do NOT collide with Player or Water
      return false;
    }

>     // Just return true if you not interested in component parent's type check result.
>     // Or call super it component's parent should be able to override the result.

Reviewer: ```suggestion
    // Just return true if you're not interested in the parent's type check result.
    // Or call super and you will be able to override the result with the parent's
    // result.
```
